lib/Sema/TypeCheckType.cpp

```diff
@@ -1979,6 +1974,16 @@ Type ResolveTypeRequest::evaluate(Evaluator &evaluator,
   if (validateAutoClosureAttributeUse(ctx.Diags, TyR, result, options))
     return ErrorType::get(ctx);
 
+  // Diagnose an attempt to use a placeholder at the top level.
+  if (isa<PlaceholderTypeRepr>(TyR) &&
```

Comment:
Does this allow `(_).self`? I think `TyR` is `TupleTypeRepr` at this point.

Reply:
Ha! Yeah, I think you're right, good catch.